components/script_traits/script_msg.rs

```diff
@@ -69,6 +69,8 @@ pub enum LogEntry {
 /// Messages from the script to the constellation.
 #[derive(Deserialize, Serialize)]
 pub enum ScriptMsg {
+    /// Forward a message to the embedder.
+    ForwardToEmbedder(EmbedderMsg),
```

Comment:
Here maybe we should have the EmbedderMsg + id?

Reply:
Yes, the thing is we still require the id inside the embedder msg, which requires this repeated call of `window.top_level_browsing_context_id`...

Reply:
How about we change `EmbedderProxy` to have a `Sender<(TopLevelBrowsingContextId, EmbedderMsg)>`(and change the relevant messages to not include the id anymore)?
That way script can just send `ScriptMsg::ForwardToEmbedder(EmbedderMsg)` without needing all those calls to `window_proxy.top_level_browsing_context_id()`, and constellation can just foward `(TopLevelBrowsingContextId, EmbedderMsg)` to the embedder?